SabreTools.Serialization/Readers/ZSTD.cs

```diff
@@ -0,0 +1,64 @@
+using System.IO;
+using SabreTools.Data.Models.ZSTD;
+using SabreTools.IO.Extensions;
+using static SabreTools.Data.Models.ZSTD.Constants;
+
+namespace SabreTools.Serialization.Readers
+{
+    public class ZSTD : BaseBinaryReader<Header>
+    {
+        /// <inheritdoc/>
+        public override Header? Deserialize(Stream? data)
+        {
+            // If the data is invalid
+            if (data == null || !data.CanRead)
+                return null;
+
+            try
+            {
+                // Cache the current offset
+                long initialOffset = data.Position;
+
+                #region Header
+                
+                var header = ParseHeader(data);
+
+                if (header == null)
+                    return null;
+                
+                #endregion
+
+                return header;
+            }
+            catch
+            {
+                // Ignore the actual error
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Parse a Stream into a Header
+        /// </summary>
+        /// <param name="data">Stream to parse</param>
+        /// <returns>Filled Header on success, null on error</returns>
+        public static Header? ParseHeader(Stream data)
+        {
+            var obj = new Header();
+            obj.VersionByte = data.ReadByte(); // ReadByte returning int is actually preferable here
```

Comment:
How is that preferable? Also, minor nitpick to add a newline above this line.

Reply:
Since I'm comparing a number. I guess it doesn't really matter; I can remove that comment if you want.

Reply:
My extension handles reading as an actual byte if that's what the value really is in a file. `ReadByteValue` is the one you'd use.

Reply:
Should be fixed.